### PR TITLE
Add flavor(variants) for stable/beta

### PR DESCRIPTION
**Pull request by PorterNan**
# What

Add flavor switch for the repo, including: 

rush switch-flavor:beta
rush switch-flavor:stable
rush update:stable
rushx current-flavor

Once run `rush switch-flavor:<stable/beta>` and `rush update<:stable>`, ou can just follow all the existing process to build the project

To try stable flavor:
`rush switch-falvor:stable` then
`rush build` or `rush test`

the only difference when you are under stable flavor is `rush update` -- run `rush update:stable` 

 
If you think you have a better name candidate, please leave a comment for those commands 😄 
# Why

# How Tested

# Process & policy checklist

- [ ] I have updated the project documentation to reflect my changes if necessary.
- [ ] I have read the [CONTRIBUTING](https://github.com/Azure/communication-ui-library/blob/main/CONTRIBUTING.md) documentation.

**Is this a breaking change?**

- [ ] This change causes current functionality to break.

### CODE REVIEW

File: common/config/env/beta.env [hunk 1]
```diff
@@ -0,0 +1 @@
+FLAVOR=beta
```
Comment (prprabhu-ms) [inline common/config/env/beta.env:1]: `FLAVOR` is too generic a name for an envvar.

Can we name this `COMMUNICATION_REACT_FLAVOR`?

File: common/config/rush/variants/stable/common-versions.json [hunk 2]
```diff
@@ -0,0 +1,65 @@
...
+   * the "dependencies" field of the top-level common/temp/package.json, which influences how the package manager
+   * will calculate versions.  The specific effect depends on your package manager.  Generally it will have no
+   * effect on an incompatible or already constrained SemVer range.  If you are using PNPM, similar effects can be
+   * achieved using the pnpmfile.js hook.  See the Rush documentation for more details.
+   *
+   * After modifying this field, it's recommended to run "rush update --full" so that the package manager
+   * will recalculate all version selections.
+   */
+  "preferredVersions": {
+    /**
```
Comment (prprabhu-ms) [inline common/config/rush/variants/stable/common-versions.json:22]: Please add a comment here describing what the value here should be (i.e., the one used in the stable build)

File: common/config/rush/variants/stable/common-versions.json [hunk 3]
```diff
@@ -0,0 +1,65 @@
...
+   * the same SemVer range for a given dependency.  However, sometimes exceptions are needed.
+   * The allowedAlternativeVersions table allows you to list other SemVer ranges that will be
+   * accepted by "rush check" for a given dependency.
+   *
+   * IMPORTANT: THIS TABLE IS FOR *ADDITIONAL* VERSION RANGES THAT ARE ALTERNATIVES TO THE
+   * USUAL VERSION (WHICH IS INFERRED BY LOOKING AT ALL PROJECTS IN THE REPO).
+   * This design avoids unnecessary churn in this file.
+   */
+  "allowedAlternativeVersions": {
+    "@azure/communication-calling": ["1.3.2"],
```
Comment (prprabhu-ms) [inline common/config/rush/variants/stable/common-versions.json:55]: Is this necessary? I thought that `allowedAlternativeVersions` is intended so that different *packlets* can use different versions at the same time.

File: common/config/rush/variants/stable/pnpmfile.js [hunk 4]
```diff
@@ -0,0 +1,38 @@
+'use strict';
+
+/**
```
Comment (prprabhu-ms) [inline common/config/rush/variants/stable/pnpmfile.js:3]: Err... why is this file needed in our case? Please remove or document.

`readPackage` doesn't have any overrides at all...

**Review by prprabhu-ms — APPROVED** [review]

**Comment by prprabhu-ms** [comment]
This is way better with fewer end-user commands per package.json

File: packages/acs-ui-common/package.json [hunk 5]
```diff
@@ -6,25 +6,24 @@
...
-    "clean": "rimraf dist",
-    "copy-original": "cpx ./src/** ./preprocessed",
-    "preprocess": "rushx copy-original && babel ./src --out-dir ../preprocessed --extensions \".ts,.tsx\" --keep-file-extension --config-file ./.babelrc.js --relative && rimraf ../preprocessed",
-    "preprocess:stable": "cross-env FLAVOR=stable rushx preprocess",
-    "clean:preprocess": "rimraf ./preprocessed",
-    "build:stable": "rushx clean:preprocess && rushx preprocess:stable && tsc -project tsconfig.preprocess.json && rushx build:cjs && api-extractor run --local",
-    "test:stable": "rushx clean:preprocess && rushx preprocess:stable && jest --verbose ./preprocessed -c jest.config.preprocess.js",
+    "clean": "rimraf dist && rimraf preprocessed",
+    "current-flavor": "echo You are running under FLAVOR: && env-cmd -f ../../common/config/env/.env node -p process.env.FLAVOR",
+    "by-flavor": "rushx current-flavor && env-cmd -f ../../common/config/env/.env --use-shell",
```
Comment (prprabhu-ms) [inline packages/acs-ui-common/package.json:15]: All the `by-flavor` variants are not intended to be run directly. Could we

* Add a prefix to all such commands, perhaps `plumbing-` or `internal-`
* Move all of them to the bottom of the `scripts`. Perhaps prefix with `zzPlumbing-`, so that sorting will naturally drop them to the bottom of the list.

File: packages/react-composites/package.json [hunk 6]
```diff
@@ -158,7 +148,7 @@
     "webpack-dev-server": "3.11.2",
     "webpack-cli": "4.7.2",
     "@babel/cli": "~7.16.0",
-    "@babel/plugin-syntax-typescript": "~7.16.0",
-    "babel-plugin-conditional-compilation": "~0.0.1"
+    "env-cmd": "~10.1.0",
```
Comment (JamesBurnside) [inline packages/react-composites/package.json:151]: nit: order deps alphabetically

File: packages/react-composites/package.json [hunk 7]
```diff
@@ -54,7 +44,7 @@
     "lint:quiet": "npm run lint -- --quiet"
   },
   "dependencies": {
-    "@azure/communication-calling": "1.3.2",
+    "@azure/communication-calling": "1.3.2-beta.1 || 1.3.2",
```
Comment (JamesBurnside) [inline packages/react-composites/package.json:47]: fyi in case this is completed first: #1294

File: common/config/rush/variants/stable/pnpmfile.js (hunk 4 above)
Comment (PorterNan) [reply common/config/rush/variants/stable/pnpmfile.js:3]: That was required by rush variant, according to their instruction:
https://rushjs.io/pages/advanced/installation_variants/#:~:text=shrinkwrap.yaml%2C%20npm%2Dshrinkwrap,as%20your%20package%20manager

File: common/config/rush/variants/stable/common-versions.json (hunk 3 above)
Comment (PorterNan) [reply common/config/rush/variants/stable/common-versions.json:55]: The rush consistent version checker only supports string comparing, so this one is a workaround - when you bump beta/stable sdk version, you need to also bump this one, so this one will be always in allow list (we have 2 different allowedVersions in stable and beta, which is good)

File: packages/react-composites/package.json (hunk 7 above)
Comment (prprabhu-ms) [reply packages/react-composites/package.json:47]: Actually, @PorterNan how will rush variants work with peerDependencies and devDepenedencies?

Thanks for linking to my PR James.

Comment (prprabhu-ms) [reply packages/react-composites/package.json:47]: My understanding is that rush variants work at the level of pnpm. So it doesn't matter why we try to install a version, it just switches out the version actually installed. But it'd be worth checking.

File: packages/acs-ui-common/package.json (hunk 5 above)
Comment (PorterNan) [reply packages/acs-ui-common/package.json]: To make it short, I've changed the name with prefix "_" (how we name internal functions and variables) and they've been moved to the bottom ;D

File: packages/react-composites/package.json (hunk 7 above)
Comment (PorterNan) [reply packages/react-composites/package.json:47]: Will merge the main branch and take a test then

File: common/config/rush/variants/stable/common-versions.json (hunk 3 above)
Comment (prprabhu-ms) [reply common/config/rush/variants/stable/common-versions.json:57]: Please add this comment in the code.

File: packages/react-composites/package.json (hunk 7 above)
Comment (PorterNan) [reply packages/react-composites/package.json]: Yea, it works when peerDependencies are stable version, but devDependencies with "1.3.2-beta.1 || 1.3.2"

**Review by JamesBurnside — APPROVED** [review]
